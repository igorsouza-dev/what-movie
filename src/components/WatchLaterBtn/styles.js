import styled from 'styled-components';

export const Container = styled.button`
  background: transparent;
  border: 0;
  padding: 5px;
  border-radius: 50%;
  position: relative;
  margin-right: 10px;
`;

export const Notification = styled.div`
  bottom: 10px;
  right: -4px;
  position: absolute;
  border-radius: 50%;
  background: #435058;
  color: #fff;
  padding: 2px;
  font-size: 10px;
  font-weight: bold;
  width: 14px;
  height: 14px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
