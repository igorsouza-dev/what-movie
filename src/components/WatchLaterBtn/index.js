import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { toggleList } from 'store/actions';
import { FaClock } from 'react-icons/fa';
import { Container, Notification } from './styles';

export default function WatchLaterBtn() {
  const dispatch = useDispatch();
  const watchLater = useSelector((state) => state.watchLater);
  const amount = Object.keys(watchLater).length;
  return (
    <Container onClick={() => dispatch(toggleList('watchLater'))} data-testid="watch-later">
      <FaClock color="#fff" size={24} />
      <Notification data-testid="count-value">{amount}</Notification>
    </Container>
  );
}
