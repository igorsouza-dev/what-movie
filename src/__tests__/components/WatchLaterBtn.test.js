import React from 'react';
import { cleanup } from '@testing-library/react';
import WatchLaterBtn from 'components/WatchLaterBtn';

import { renderWithRedux } from '../utils/redux';

describe('<WatchLaterBtn />', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container, getByTestId } = renderWithRedux(
      <WatchLaterBtn />
    );

    const linkElement = getByTestId(/watch-later/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const { getByTestId } = renderWithRedux(
      <WatchLaterBtn />,
      {
        initialState: {
          watchLater: {
            1234: { id: 1234, original_title: 'Test 1' },
            1235: { id: 1235, original_title: 'Test 2' },
          },
        },
      }
    );

    expect(getByTestId(/count-value/i)).toHaveTextContent('2');
  });
});
