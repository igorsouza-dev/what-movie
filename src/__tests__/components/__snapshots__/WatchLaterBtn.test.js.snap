// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchLaterBtn /> should render 1`] = `
<button
  class="sc-bdVaJa eXEdri"
  data-testid="watch-later"
>
  <svg
    color="#fff"
    fill="currentColor"
    height="24"
    size="24"
    stroke="currentColor"
    stroke-width="0"
    style="color: rgb(255, 255, 255);"
    viewBox="0 0 512 512"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
    />
  </svg>
  <div
    class="sc-bwzfXH buZGzN"
    data-testid="count-value"
  >
    0
  </div>
</button>
`;
